platform/mbed_application.c

```diff
@@ -158,7 +161,11 @@ static void powerdown_scb(uint32_t vtor)
 
 __asm static void start_new_application(void *sp, void *pc)
 {
+#if defined(__CORTEX_M0PLUS)
```

Comment:
It's best to just use MOVS unconditionally here - it's a 16-bit instruction instead of 32-bit (which is why the M0+ supports it). So you'll save ROM on all Cortex-Ms.

Reply:
Good call. I've added this in my most recent commit.